include/wx/image.h

```diff
@@ -628,6 +651,46 @@ class WXDLLIMPEXP_CORE wxImage: public wxObject
     // modified versions of this image.
     wxImage MakeEmptyClone(int flags = Clone_SameOrientation) const;
 
+    // Helper function used internally by wxImage class only.
+    template <typename T>
+    void ApplyToAllPixels(void (wxImage::*func)(unsigned char *, T), T value);
```

Comment:
It seems to me that none of the functions used with this one actually needs to be a member function, so I'd use
```suggestion
    void ApplyToAllPixels(void (*filter)(const RGBValue* mask, unsigned char *, T), T value);
```
here and just pass it the mask RGB (the pointer would be null if there is no mask).

I suspect that it would be beneficial to have 2 different execution paths/loops for the case with/without mask as the latter should be quite a bit faster as we don't need any tests at all inside the loop, and so it should be auto-vectorized, but this would require more changes.

Reply:
I'll move them to the source file, and maybe make `ApplyToAllPixels` `protected` to allow sub-classing customization.
the mask condition check `if ( !HasMask() || rgb[0] != GetMaskRed() || rgb[1] != GetMaskGreen() || rgb[2] != GetMaskBlue() )` is called for each pixel and is composed from **OR**s.